doc/link-expansion.md

```diff
@@ -0,0 +1,287 @@
+      "links": {}
+    }]
+  },
+}
+```
+
+Within a `links` json object there are keys which indicate which a type of link,
+and at the value of those keys is an array of all links of that type. In the
+above example there are two types of link: `organisations` and
+`available_translations` which contain 1 and 2 links respectively.
+
+## When it occurs
+
+Link expansion occurs at the point a content item is represented to the
+Content Store - which is normally the result of a [Sidekiq](http://sidekiq.org/)
+worker process such as [DownstreamDraftWorker][downstream-draft-worker] or
+[DownstreamLiveWorker][downstream-live-worker].
+
+## Link sources
+
+The links for a content item are a combination of links which are added
+via [patch-link-set](api.md#patch-v2linkscontent_id) and those that are
+determined automatically.
+
+### Links added via patch-link-set (dependees)
+
+Links added via patch-link-set are stored for a content item as a
+collection of content ids, all with a link type and ordering index. These are
+always presented as part of link expansion, and some of the link_types provided
+here may also include the links for the linked item if they are of a
+[recursive link type](#recursive-link-type).
+
+These are considered to be **depenees** of a content item and are determined
+using the [Presenters::Queries::ExpandDependees][expand-dependees] class.
+
+### Links added automatically
+
+There are two types of links which are added automatically as part of link
+expansion, these are reverse links and available translations.
+
+#### Reverse links (dependents)
+
+Some types of links are presented when they are the target of a link.
+
+Consider item `A` which links to item `B` - with type `reciprocal` - and `B` does
+not have a link to `A`. Under normal circumstances this will mean that the
+links for `A` would include `B`, yet `B` would not include `A` in it's
+links. However if `reciprocal` is defined as a reverse link it will also be
+included in the links for `B`, under a link type of the defined reverse name
+for "reciprocal". These are defined in
+[Queries::DependentExpansionRules][dependent-expansion-rules].
+
+An example of a reverse link type is `parent` which is defined as having a
+reverse name of `children`. Consider item `C` which has a link to
+item `D` with type `parent`. In `C` you would have a group of `parents`
+links which includes `D`, whereas in `D` you would have a group of
+`children` links which includes `C`.
+
+Reverse links are considered to be **dependents** of a content item and are
+determined using the [Presenters::Queries::ExpandDependents][expand-dependents]
+class.
+
+#### Available translations
+
+A content item can be available in multiple translations. This will be
+determined by content items sharing the same [content_id][content-id] and
+having different [locale][locale] values.
+
+The links for a content item will include all the translations of a content
+item, including the current locale.
+
+**eg**
```

Comment:
Probably being really pedantic, but I would write ‘Example’ here just to make it 100% clear.